PABC.Server/Features/GetApplicationRolesPerEntityType/GetApplicationRolesPerEntityTypeController.cs

```diff
@@ -1,58 +1,62 @@
+    [ApiController]
+    [Route("/api/v1/application-roles-per-entity-type")]
+    [Authorize(Policy = ApiKeyAuthentication.Policy)]
+    public class GetApplicationRolesPerEntityTypeController(PabcDbContext db) : ControllerBase
     {
-        var query = db.Mappings
-            .Where(x => request.FunctionalRoleNames.Contains(x.FunctionalRole.Name)) // search for matching function roles, unknown function roles get ignored
-            .SelectMany(x => x.Domain.EntityTypes, (m, e) => new { ApplicationRoleName = m.ApplicationRole.Name, m.ApplicationRole.Application, e.Id, e.Type, e.EntityTypeId, EntityTypeName = e.Name });
+        [HttpPost(Name = "Get application roles per entity type")]
+        [Consumes(MediaTypeNames.Application.Json)]
+        [ProducesResponseType<GetApplicationRolesResponse>(StatusCodes.Status200OK, MediaTypeNames.Application.Json)]
+        [ProducesResponseType<ValidationProblemDetails>(StatusCodes.Status400BadRequest,
+            MediaTypeNames.Application.ProblemJson)]
+        [ProducesResponseType<ProblemDetails>(StatusCodes.Status401Unauthorized,
+            MediaTypeNames.Application.ProblemJson)]
+        public async Task<ActionResult<GetApplicationRolesResponse>> Post([FromBody] GetApplicationRolesRequest request,
+            CancellationToken token = default)
+        {
+            var allEntityTypesQuery = from m in db.Mappings
+                    .Include(m => m.ApplicationRole)
+                    .Include(m => m.FunctionalRole)
+                    .Include(m => m.Domain)
+                    .Where(x => request.FunctionalRoleNames.Contains(x.FunctionalRole.Name) && x.IsAllEntityTypes)
+                from e in db.EntityTypes
+                select new { m, e };
 
-        var result = new Dictionary<Guid, GetApplicationRolesResponseModel>();
+            var domainSpecificQuery = from m in db.Mappings
+                    .Include(m => m.ApplicationRole)
+                    .Include(m => m.FunctionalRole)
+                    .Include(m => m.Domain)
+                    .Where(x => request.FunctionalRoleNames.Contains(x.FunctionalRole.Name) && !x.IsAllEntityTypes)
+                from e in m.Domain!.EntityTypes
+                select new { m, e };
 
-        await foreach (var item in query.AsAsyncEnumerable())
-        {
-            if (!result.TryGetValue(item.Id, out var val))
-            {
-                val = new()
-                {
-                    ApplicationRoles = [],
-                    EntityType = new()
-                    {
-                        Id = item.EntityTypeId,
-                        Type = item.Type,
-                        Name = item.EntityTypeName,
-                    }
-                };
-                result.Add(item.Id, val);
-            }
+            var combinedQuery = allEntityTypesQuery.Union(domainSpecificQuery);
```

Comment:
I think this will result in quite a complex sql query. Could we maybe just fetch all entity types in advance, in a separate query?